sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -0,0 +1,171 @@
+    `<tempfile.gettempdir()>/Microsoft/AzureMonitor/opentelemetry-python-<your-instrumentation-key>`.
+    :rtype: None
+    """
+
+    configurations = _get_configurations(**kwargs)
+
+    disable_tracing = configurations[DISABLE_TRACING_ARG]
+    disable_logging = configurations[DISABLE_LOGGING_ARG]
+    disable_metrics = configurations[DISABLE_METRICS_ARG]
+
+    # Setup tracing pipeline
+    if not disable_tracing:
+        _setup_tracing(configurations)
+
+    # Setup logging pipeline
+    if not disable_logging:
+        _setup_logging(configurations)
+
+    # Setup metrics pipeline
+    if not disable_metrics:
+        _setup_metrics(configurations)
+
+    # Setup instrumentations
+    # Instrumentations need to be setup last so to use the global providers
+    # instanstiated in the other setup steps
+    _setup_instrumentations(configurations)
+
+
+def _setup_tracing(configurations: Dict[str, ConfigurationValue]):
+    sampling_ratio = configurations[SAMPLING_RATIO_ARG]
+    tracer_provider = TracerProvider(
+        sampler=ApplicationInsightsSampler(sampling_ratio=cast(float, sampling_ratio)),
+    )
+    set_tracer_provider(tracer_provider)
+    trace_exporter = AzureMonitorTraceExporter(**configurations)
+    span_processor = BatchSpanProcessor(
+        trace_exporter,
+    )
+    get_tracer_provider().add_span_processor(span_processor)
+
+
+def _setup_logging(configurations: Dict[str, ConfigurationValue]):
+    # TODO: Remove after upgrading to OTel SDK 1.18
+    logging_export_interval_ms = configurations[LOGGING_EXPORT_INTERVAL_MS_ARG]
+    logger_provider = LoggerProvider()
+    set_logger_provider(logger_provider)
+    log_exporter = AzureMonitorLogExporter(**configurations)
+    log_record_processor = BatchLogRecordProcessor(
+        log_exporter,
+        schedule_delay_millis=cast(int, logging_export_interval_ms),
+    )
+    get_logger_provider().add_log_record_processor(log_record_processor)
+    handler = LoggingHandler(logger_provider=get_logger_provider())
+    getLogger().addHandler(handler)
+
+
+def _setup_metrics(configurations: Dict[str, ConfigurationValue]):
+    metric_exporter = AzureMonitorMetricExporter(**configurations)
+    reader = PeriodicExportingMetricReader(metric_exporter)
+    meter_provider = MeterProvider(
+        metric_readers=[reader],
+    )
+    set_meter_provider(meter_provider)
+
+
+def _setup_instrumentations(configurations: Dict[str, ConfigurationValue]):
+    disabled_instrumentations = configurations[DISABLED_INSTRUMENTATIONS_ARG]
+
+    # use pkg_resources for now until https://github.com/open-telemetry/opentelemetry-python/pull/3168 is merged
+    for entry_point in iter_entry_points(
+        "azure_monitor_opentelemetry_instrumentor"
+    ):
+        lib_name = entry_point.name
+        if lib_name not in _SUPPORTED_INSTRUMENTED_LIBRARIES_DEPENDENCIES_MAP:
+            continue
+        if entry_point.name in disabled_instrumentations:
```

Comment:
@annatisch You can now disable instrumentations using the OTEL_PYTHON_DISABLED_INSTRUMENTATIONS env var